pkg/kotsutil/kots.go

```diff
@@ -163,6 +165,56 @@ func (k *KotsKinds) HasPreflights() bool {
 	return len(k.Preflight.Spec.Analyzers) > 0
 }
 
+func HasStrictPreflights(preflight *troubleshootv1beta2.Preflight) (bool, error) {
+	if preflight == nil {
+		return false, nil
+	}
+
+	marshalledAnalyzers, err := json.Marshal(preflight.Spec.Analyzers) // marshall and remove nil Analyzers eg result: "[{\"clusterVersion\":{\"exclude\":\"\",\"strict\":\"false\",\"outcomes\":null}}]"
+	if err != nil {
+		return false, errors.Wrap(err, "error while marshalling preflight.Spec.Analyzers")
+	}
+
+	analyzersMap := []map[string]interface{}{}
+	err = json.Unmarshal(marshalledAnalyzers, &analyzersMap) // Unmarshall again so we can loop over non nil analyzers
+	if err != nil {
+		return false, errors.Wrap(err, "error while un-marshalling marshalledAnalyzers")
+	}
+
+	// analyzerMap will ignore empty Analyzers and loop around Analyzer with data
+	for _, analyzers := range analyzersMap { // for each analyzers: map["clusterVersion": map[string]interface{} ["exclude": "", "strict": "true", "outcomes": nil]
```

Comment:
about the example in the comment, this go format is hard to parse, can you just post the example in json?
```suggestion
	for _, analyzers := range analyzersMap { // for each analyzer: map["clusterVersion": map[string]interface{} ["exclude": "", "strict": "true", "outcomes": nil]
```